server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -308,14 +308,16 @@ private void recoverSeqFiles(List<TsFileResource> tsFiles) throws StorageGroupPr
       TsFileRecoverPerformer recoverPerformer = new TsFileRecoverPerformer(storageGroupName + "-"
           , schema, versionController, tsFileResource, false, i == tsFiles.size() - 1);
       RestorableTsFileIOWriter writer = recoverPerformer.recover();
-      if (i != tsFiles.size() - 1) {
+      if (i != tsFiles.size() - 1 || !writer.canWrite()) {
         // not the last file, just close it
```

Comment:
change the comment, what  !write.canWrite means?

Reply:
Fixed.